core/src/main/scala/kafka/server/ConfigHelper.scala

```diff
@@ -96,11 +81,21 @@ class ConfigHelper(metadataCache: MetadataCache, config: KafkaConfig, configRepo
                       includeSynonyms: Boolean,
                       includeDocumentation: Boolean): List[DescribeConfigsResponseData.DescribeConfigsResult] = {
     resourceToConfigNames.map { resource =>
-
-      def createResponseConfig(configs: java.util.Map[String, _ <: Object],
+      def createResponseConfig(configs: Any,
                                createConfigEntry: (String, Object) => DescribeConfigsResponseData.DescribeConfigsResourceResult): DescribeConfigsResponseData.DescribeConfigsResult = {
+        val stream = configs match {
+          case c: AbstractConfig =>
+            java.util.stream.Stream.concat(
+              c.originals.entrySet.stream.filter(_.getValue != null),
+              c.nonInternalValues.entrySet.stream.map(_.asInstanceOf[java.util.Map.Entry[String, Object]])
```

Comment:
You might change the type of `createConfigEntry` from `(String, Object)` to `(String, Any)` to eliminate the extra type declaration.